src/workerd/api/web-socket.h

```diff
@@ -146,6 +146,37 @@ class ErrorEvent: public Event {
   void visitForGc(jsg::GcVisitor& visitor);
 };
 
+class WebSocketProtocolError {
+public:
+  WebSocketProtocolError() = default;
+  WebSocketProtocolError(int code, kj::String description)
+      : code(code), description(kj::mv(description)) {}
+
+  int getCode() const { return code; }
+  kj::StringPtr getDescription() const { return description; }
+
+  [[nodiscard]] bool decodeFromException(const kj::Exception& ex);
+  // Fills out this object from the exception's context, but only if the context actually holds
+  // appropriate data. Returns true if successful, false otherwise.
+
+  void encodeToException(kj::Exception& ex) &&;
+  // Adds a context entry to the exception containing this object's data. This is only useful if
+  // you're going to retrieve it with decodeFromException later.
+
+private:
+  int code;
+  kj::String description;
+  static inline constexpr char magicFileValue[] = "__WebSocketProtocolError_magicFileValue";
```

Comment:
Non-blocking nit: can you add a short comment here explaining what `magicFileValue` is for?

Also, this could likely be a constexpr kj::StringPtr instead of char array.

Reply:
Done.